spec/CRM/Example/RiskManager/Projection.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+-- https://downloads.haskell.org/ghc/latest/docs/users_guide/using-warnings.html#ghc-flag--Wmissing-deriving-strategies
+{-# OPTIONS_GHC -Wno-missing-deriving-strategies #-}
+
+module CRM.Example.RiskManager.Projection where
+
+import "crm" CRM.BaseMachine
+import CRM.Example.RiskManager.Domain
+import "crm" CRM.Topology
+import "base" Data.Monoid (Last (..))
+import "singletons-base" Data.Singletons.Base.TH
+import "base" GHC.Generics (Generic)
+
+data ReceivedData = ReceivedData
+  { receivedUserData :: Maybe UserData
+  , receivedLoanDetails :: Maybe LoanDetails
+  , receivedCreditBureauData :: Maybe CreditBureauData
+  }
+  deriving stock (Eq, Show, Generic)
+
+instance Semigroup ReceivedData where
+  (<>) :: ReceivedData -> ReceivedData -> ReceivedData
+  (<>) (ReceivedData ud1 ld1 cbd1) (ReceivedData ud2 ld2 cbd2) =
+    ReceivedData
+      (getLast $ Last ud1 <> Last ud2)
+      (getLast $ Last ld1 <> Last ld2)
+      (getLast $ Last cbd1 <> Last cbd2)
```

Comment:
```suggestion
r1 <> r2 = ReceivedData
    { receivedUserData = receivedUserData r1 <|> receivedUserData r2
    , receivedLoanDetails = receivedLoanDetails r1 <|> receivedLoanDetails r2
    , receivedCreditBureauData = receivedCreditBureauData r1 <|> receivedCreditBureauData r2
    }
```

Reply:
makes sense, but it still needs the `Last` and `getLast`